utils/zerovec/derive/src/varule.rs

```diff
@@ -110,3 +114,399 @@ pub fn derive_impl(input: &DeriveInput) -> TokenStream2 {
+    /// str
+    Str,
+}
+
+/// Represents the type of the last field of the struct
+#[derive(Clone, Eq, PartialEq)]
+enum LastField<'a> {
+    Cow(OwnULETy<'a>),
+    ZeroVec(&'a Type),
+    VarZeroVec(&'a Type),
+
+    // Generally you should be using the above ones for maximum zero-copy, but these will still work
+    Growable(OwnULETy<'a>),
+    Boxed(OwnULETy<'a>),
+    Ref(OwnULETy<'a>),
+}
+
+impl<'a> LastField<'a> {
+    /// Construct a LastField for the type of a LastField if possible
+    fn new(ty: &'a Type) -> Result<LastField<'a>, String> {
+        match *ty {
+            Type::Reference(ref tyref) => OwnULETy::new(&tyref.elem, "reference").map(LastField::Ref),
+            Type::Path(ref typath) => {
+                if typath.path.segments.len() != 1 {
+                    return Err("Can only automatically detect corresponding VarULE types for \
+                                path types with a single path segment".into());
+                }
+                let segment = typath.path.segments.first().unwrap();
+                match segment.arguments {
+                    PathArguments::None => {
+                        if segment.ident == "String" {
+                            Ok(LastField::Growable(OwnULETy::Str))
+                        } else {
+                            Err("Can only automatically detect corresponding VarULE types for path types \
+                                        that are not Cow, ZeroVec, VarZeroVec, Box, String, or Vec".into())
+                        }
+                    }
+                    PathArguments::AngleBracketed(ref params) => {
+                        // At most one lifetime and exactly one generic parameter
+                        let mut lifetime = None;
+                        let mut generic = None;
+                        for param in &params.args {
+                            match param {
+                                GenericArgument::Lifetime(ref lt) if lifetime.is_none() => {
+                                    lifetime = Some(lt)
+                                }
+                                GenericArgument::Type(ref ty) if generic.is_none() => {
+                                    generic = Some(ty)
+                                }
+                                _ => return Err("Can only automatically detect corresponding VarULE types for path \
+                                                 types with at most one lifetime and at most one generic parameter".into()),
+                            }
+                        }
+
+                        // Must be exactly one generic parameter
+                        let generic = if let Some(g) = generic {
+                            g
+                        } else {
+                            return Err("Can only automatically detect corresponding VarULE types for path \
+                                        types with at most one lifetime and at most one generic parameter".into());
```

Comment:
```suggestion
                                        types with at most one lifetime and exactly one generic parameter".into());
```

Reply:
No, `str` is a path type without one generic